R/LSproblemofpointscommon.R

```diff
@@ -1,227 +1,234 @@
+# y: a vector containing the number of trials each player needs to win
+# a: a vector containing all the parameters for the Dirichlet multinomial distribution, indicating the competence of each player
+
+pDirichletNegMultinom <- function(y, a){
+  dDirichletNegMultinom <- function(y, a){
     #y <- c(4,3,2)
     #a <- c(2,3,4)
-    y_sum <- sum(y)
-    a_sum <- sum(a)
-    p1 <- gamma(y_sum) / (gamma(y[1]) * prod(gamma(y+1))/gamma(y[1]+1))
-    p2 <- gamma(a_sum) / (prod(gamma(a)))
-    p3 <- (prod(gamma(y + a))) / gamma(y_sum + a_sum)
+    ySum <- sum(y)
+    aSum <- sum(a)
+    p1 <- gamma(ySum) / (gamma(y[1]) * prod(gamma(y+1))/gamma(y[1]+1))
+    p2 <- gamma(aSum) / (prod(gamma(a)))
+    p3 <- (prod(gamma(y + a))) / gamma(ySum + aSum)
     return (p1 * p2 * p3)
   }
 
   #calculate the prob
-  p_calculated <- 0
+  pCalculated <- 0
   for (l in 1:(prod(y)/y[1])){         #generate all the situations that player 1 wins
-    p_calculated <- p_calculated + dDirichletnmultinom(combinations(y)[l,], a)
+    pCalculated <- pCalculated + dDirichletNegMultinom(combinations(y)[l, ], a)
   }
-  return(p_calculated)
+  return(pCalculated)
 }
 
+
+
+# This is a recursive function returning all situations (combinations of points obtained 
+# by each player) in which player 1 wins the game.
+# Input k: a vector of containing the number of trials that each player needs to win the game.
+
 combinations <- function(k){
-  if (length(k) == 3){ # the number of players
-    #k <- c(1,2,3)
+  n <- length(k)
+  
+  if (n == 3){ # the number of players
     num <- prod(k)/k[1] #total number of sets
-    #sets <- vector(mode = "list", length = num)
-    sets <- matrix(nrow = num, ncol = length(k))
+    sets <- matrix(nrow = num, ncol = n)
     for (i in 0:(k[2]-1)){
       for (j in 0:(k[3]-1)){
-        sets[i*k[3]+j+1,] <- c(k[1],i,j)
+        sets[i*k[3]+j+1, ] <- c(k[1],i,j)
       }
     }
     return(sets)
   }else{
-    #k <- c(1,2,3,4)
-    #x1 <- sets
-    x1 <- combinations(k[1:(length(k) - 1)]) # k without k[length(k)]
-    x2 <- k[length(k)]
-    y <- matrix(ncol = length(k), nrow = prod(k)/k[1])
-    for (i in 1:length(x1[,1])){
-      for (j in 1:k[length(k)]){
-        y[(i-1)*(k[length(k)])+j,] <- append(x1[i,],j-1)
+    x1 <- combinations(k[1:(n-1)]) # recursion
+    x2 <- k[n]
+    sets <- matrix(ncol = n, nrow = prod(k)/k[1])
+    for (i in 1:length(x1[ ,1])){
+      for (j in 1:k[n]){
+        sets[(i-1)*(k[n])+j, ] <- append(x1[i, ],j-1)
       }
     }
-    return(y)
+    
+    return(sets)
   }
 }
 
-compare_function1 <- function(p,m1,n1,t,simulation){
+compareChanceTwoPlayers <- function(p,m1,n1,t,simulation){
 
   # first estimating the probability that player 1 wins by simulating the game for a given number of times.
   m <- t - m1 # m: the number of success that player 1 should have to win the game
   n <- t - n1 # n: the number of success that player 2 should have to win the game
 
-  record_game <- vector()
+  recordGame <- vector()
```

Comment:
```suggestion
  recordGame <- numeric(simulation)
```
or `integer(simulation)` if you prefer to be pedantic about the types.